partner_communication_revision/models/communication_revision.py

```diff
@@ -234,6 +239,19 @@ def write(self, vals):
         Push back the enhanced text in translation of the mail.template.
         Update revision date and number depending on edit mode.
         """
+        if "active_rev_history_id" in vals and self.active_rev_history_id:
+            self.active_rev_history_id.save_revision_state()
+            backup = self.env["partner.communication.revision.history"]\
+                .sudo().browse(vals["active_rev_history_id"])
```

Comment:
```suggestion
                .browse(vals["active_rev_history_id"])
```
sudo should not be required here